model.rb

```diff
@@ -124,14 +124,26 @@ def redacted_columns
 end
 
 module HealthMonitorMethods
-  def aggregate_readings(previous_pulse:, reading:, data: {})
+  def aggregate_readings(previous_pulse: {}, reading:, data: {})
+    previous_pulse ||= {}
     {
       reading: reading,
       reading_rpt: (previous_pulse[:reading] == reading) ? previous_pulse[:reading_rpt] + 1 : 1,
       reading_chg: (previous_pulse[:reading] == reading) ? previous_pulse[:reading_chg] : Time.now
     }.merge(data)
   end
 
+  # There is some code duplication here but we might have different signals
+  # which could need its own aggregation. So keeping aggregate logic separate.
+  def aggregate_disk_readings(previous_disk_health:, reading:, data: {})
+  previous_disk_health ||= {}
```

Comment:
What's a case we would need different aggregation logic? For now, it looks like we don't need any. So, let's reduce the code duplication here.

Reply:
you can also create a single function named `basic_aggregation`, and if the next signal can use it, it would. Otherwise, they would implement their own.

Reply:
Yeah, for network we have different aggregation logic.
Will see what I can do here.

Reply:
@furkansahin Would like to leave this as is, not gaining much.
The idea was we can have different logging frequencies and if disk also involves other signals of utilisation or temperature, it might be handy.